### PR TITLE
Initial auth setup dialog

### PR DESCRIPTION
**Pull request by rli**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
image

image

image

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildSrc/src/main/kotlin/toolkit-intellij-subplugin.gradle.kts [hunk 1]
```diff
@@ -169,6 +169,7 @@ tasks.runIde {
     systemProperty("aws.toolkit.developerMode", true)
     systemProperty("ide.plugins.snapshot.on.unload.fail", true)
     systemProperty("memory.snapshots.path", project.rootDir)
+    systemProperty("idea.auto.reload.plugins", false)
```
Comment (manodnyab) [inline buildSrc/src/main/kotlin/toolkit-intellij-subplugin.gradle.kts:172]: What is this required for?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+                        model.addElement(it)
+                    }
+            }
+
+            cell(combo)
+                .bindItem(state.idcTabState::roleInfo)
+        }
+    }
+
+    private fun builderIdTab() = panel {
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt:303]: Is there a way to have a common function for these panels?

File: buildSrc/src/main/kotlin/toolkit-intellij-subplugin.gradle.kts (hunk 1 above)
Comment (rli) [reply buildSrc/src/main/kotlin/toolkit-intellij-subplugin.gradle.kts:172]: stops annoying plugin reloading all the time

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt (hunk 2 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt:303]: what would be the common part?

Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt:316]: Nvm i got confused with the other tabs.

**Review by manodnyab — COMMENTED** [review]
Can you add screenshots?

**Review by manodnyab — APPROVED** [review]

**Pull request merged**